crates/llvm-context/src/polkavm/context/pointer.rs

```diff
@@ -76,6 +76,35 @@ impl<'ctx> Pointer<'ctx> {
         Self::new(r#type, address_space, value)
     }
 
+    /// Creates a new pointer with the specified `offset` for storage operations.
+    pub fn new_with_store_offset<D, T>(
+        context: &Context<'ctx, D>,
+        address_space: AddressSpace,
+        r#type: T,
```

Comment:
This is not needed. We always return a register sized (xlen) pointer (because it contains the allocation pointer and no the storage pointer).